code/modules/antagonists/wizard/wizard.dm

```diff
@@ -224,6 +224,20 @@
 			owner.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/knock(null))
 			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/mind_transfer(null))
 			to_chat(owner, "<B>Your service has not gone unrewarded, however. Studying under [master.current.real_name], you have learned stealthy, robeless spells. You are able to cast knock and mindswap.")
+		if(APPRENTICE_WILDMAGIC)
+			var/static/list/spell_lists = subtypesof(/datum/spellbook_entry)-typesof(/datum/spellbook_entry/item)-typesof(/datum/spellbook_entry/summon)
+			var/spells_left = 2
+			while(spells_left)
+				var/datum/spellbook_entry/chosen_spell = pick(spell_lists)
+				for(var/obj/effect/proc_holder/spell/my_spell in owner.spell_list)
+					if(initial(chosen_spell.name) == initial(my_spell.name)) // You don't learn the same spell
+						continue
+					if(is_type_in_typecache(my_spell, initial(chosen_spell.no_coexistance_typecache))) // You don't learn a spell that isn't compatible with another
+						continue
```

Comment:
This won't properly prevent you from getting the same spell twice, or an incompatible one

Reply:
Ah, right. Bad loop.